src/System.Text.Json/src/System/Text/Json/JsonElement.cs

```diff
@@ -0,0 +1,372 @@
+
+            if (rawData.Length > destination.Length)
+            {
+                bytesWritten = 0;
+                return false;
+            }
+
+            rawData.Span.CopyTo(destination);
+            bytesWritten = rawData.Length;
+            return true;
+        }
+
+        public bool GetBoolean()
+        {
+            JsonTokenType type = Type;
+
+            return
+                type == JsonTokenType.True ? true :
+                type == JsonTokenType.False ? false :
+                throw new InvalidOperationException();
+        }
+
+        public string GetString()
+        {
+            if (_parent == null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            return _parent.GetString(_idx);
+        }
+
+        public bool TryGetValue(out int value)
+        {
+            if (_parent == null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            return _parent.TryGetValue(_idx, out value);
+        }
+
+        public int GetInt32()
+        {
+            if (TryGetValue(out int value))
+            {
+                return value;
+            }
+
+            throw new FormatException();
+        }
+
+        public bool TryGetValue(out long value)
+        {
+            if (_parent == null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            return _parent.TryGetValue(_idx, out value);
+        }
+
+        public long GetInt64()
+        {
+            if (TryGetValue(out long value))
+            {
+                return value;
+            }
+
+            throw new FormatException();
+        }
+
+        [CLSCompliant(false)]
+        public bool TryGetValue(out ulong value)
+        {
+            if (_parent == null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            return _parent.TryGetValue(_idx, out value);
+        }
+
+        [CLSCompliant(false)]
+        public ulong GetUInt64()
+        {
+            if (TryGetValue(out ulong value))
+            {
+                return value;
+            }
+
+            throw new FormatException();
+        }
+
+        public bool TryGetValue(out double value)
+        {
+            if (_parent == null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            return _parent.TryGetValue(_idx, out value);
+        }
+
+        public double GetDouble()
+        {
+            if (TryGetValue(out double value))
+            {
+                return value;
+            }
+
+            throw new FormatException();
+        }
+
+        public ChildEnumerator EnumerateChildren()
+        {
+            if (_parent == null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            return new ChildEnumerator(this);
+        }
+
+        public override string ToString()
```

Comment:
I'm not sure of the scenarios for this, but if they exist, would there be any benefit to a TryFormat in addition to ToString?

Reply:
All Claims/JWT does with the JSON is ToString (into their existing model that needs System.String), so this was a direct customer request.

TryFormat feels weird, since we'd have to get the correct interpreted value and delegate... it'd be better for the caller to get the correct interpreted value so that if they have to call TryFormat twice we didn't have to parse twice.